hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1081,14 +1088,19 @@ bool MDMParser::init(DevStatus* status)
                 if (RESP_OK != waitFinalResp(nullptr, nullptr, CFUN_TIMEOUT)) {
                     goto failure;
                 }
+                waitFinalResp(nullptr, nullptr, 1000); // delay and pump URCs a bit to wait for disconnect
             }
             int p = UBLOX_SARA_UMNOPROF_SIM_SELECT;
             if (netProv == CELLULAR_NETPROV_TWILIO) {
                 p = UBLOX_SARA_UMNOPROF_STANDARD_EUROPE;
             }
-            sendFormated("AT+UMNOPROF=%d\r\n", p);
-            waitFinalResp(nullptr, nullptr, UMNOPROF_TIMEOUT);
-            goto reset_failure; // Not checking for errors above since we will reset either way
+            // _oldFirmwarePresent: u-blox firmware 05.06* and 05.07* does not have
+            // UMNOPROF=100 available. Default to UMNOPROF=0 in that case.
+            if (!(_oldFirmwarePresent && p == UBLOX_SARA_UMNOPROF_STANDARD_EUROPE)) {
+                sendFormated("AT+UMNOPROF=%d\r\n", p);
+                waitFinalResp(nullptr, nullptr, UMNOPROF_TIMEOUT);
+                goto reset_failure; // Not checking for errors above since we will reset either way
+            }
```

Comment:
Wouldn't it be better to avoid setting `CFUN=0` as well if the modem is running an old firmware?

Reply:
Good call, I'll make that change.